src/Makefile.am

```diff
@@ -159,3 +161,14 @@ apps_plugin_LDADD = \
 	$(OPTIONAL_CAP_LIBS) \
 	$(NULL)
 
+if FREEIPMI
+plugins_PROGRAMS += freeipmi.plugin
```

Comment:
only this should be in conditional, leaving the rest unconditioned, this will work as their target will not pulled in.
so you have something like:
```
plugins_PROGRAMS =
if X
plugins_PROGRAMS +=X
endif
if Y
plugins_PROGRAMS += Y
endif

# continue as usual with no conditionals
```